contracts/Wrapped1155Factory.sol

```diff
@@ -135,16 +145,32 @@ contract Wrapped1155Factory is ERC1155Receiver {
             
             // assign name
             hex"7f",
-            "Wrapped ERC-1155", uint128(32),
+            // Chars in hex or UTF-8 = 8 bits => 2 hex numbers => 
+            // string + uint128(30) for 16 chars needs to padding zeros 
+            // and the length of the string
+            // leading zeros = (64 - (tokenName.length * 2)) / 2
+            // Wrapped ERC-1155GnosisProtocol = 31 * 2 zeros + length * 2 in hex
+            // (62 - (31 * 2)) = 0 zeros + length * 2 in hex
+            // "Wrapped ERC-1155", uint128(32), 
+            // Wrapped ERC-1155 = 16 zeros + length * 2 in hex
+            // WrappedERC-1155 = 15 * 2 zeros + length * 2 in hex
+            // (62 - (15 * 2)) = 16
+            //"WrappedERC-1155", hex"000000000000000000000000000000001E", 
+            tokenName, tokenNameOffset, 
             hex"600655",
             
             // assign symbol
             hex"7f",
+            // Chars in hex or UTF-8 = 8 bits => 2 hex numbers => 
+            // string + uint128(30) for 16 chars needs to padding zeros 
+            // and the length of the string
             "WMT", uint232(6),
```

Comment:
Instead of computing these offsets by hand, maybe a helper function that computes this "string-31" encoding might be helpful:
```solidity
    function toString31(string memory value) public pure returns (bytes32 encodedString) {
        uint256 length = bytes(value).length;
        require(length < 32, "string too long");
        
        // Read the right-padded string data, which is guaranteed to fit into a single
        // word because its length is less than 32.
        assembly {
            encodedString := mload(add(value, 0x20))
        }
        
        // Now mask the string data, this ensures that the bytes past the string length
        // are all 0s.
        bytes32 mask = bytes32(uint256(-1) << ((32 - length) << 3));
        encodedString = encodedString & mask;
        
        // Finally, set the least significant byte to be the hex? length of the encoded
        // string, that is its byte-length times two.
        encodedString = encodedString | bytes32(length << 1);
    }
```

I just checked and hist produces `0x577261707065644552432d31313535000000000000000000000000000000001e` for "WrappedERC-1155".

Reply:
Thank you @nlordell ! I wanted to use another function to do a similar convert from optional addresses from a `bytes calldata _payload`. 

For the token info we wanted to make it cost efficient, if we could send the bytecode already generated with the `bytes calldata data` parameter maybe we could safe come gas. That's the reason because I computed the token name, symbol, and decimals. 

I created a helper just to calculate the token bytecode. What do you think?

Reply:
Not something for now, but one way to improve readability of the contract would be:

1. Instead of generating the code directly through EVM opcodes, use the clone function from Openzeppelin framework. https://docs.openzeppelin.com/contracts/3.x/api/proxy#Clones


My doubt would be how to set up the variables tokenName, symbol, decimals... I image they just need to be defined in the contract layout in the same order and with the same types thant the mastercopy of the proxy.

Reply:
actually that's the limitation from that library, you cannot add variables to the proxy call, that's why alan chose the evm option directly, it's the only way I know to use the minimal proxy thing for now with variables

Reply:
Let's see in the future the proxy Clone